### ISSUE TITLE
Ask if scenario should be started as network game when opeing c4Scenario file

### ISSUE THREAD

**Issue by Somebodyisnobody** [legacyclonk/LegacyClonk#86]
When double clicking on a .c4s file clonk starts the scenario without network as singleplayer session with currently activated players.
I suggest to ask the user over a dialog wether

1. the game should start in network mode and a lobby should be opened or
2. the game should be started in singleplayer / splitscreen without network players.

**Comment by Fulgen301** [comment]
This isn't an engine issue, but an issue of starting the engine with the correct arguments, which would require multiple explorer shell entries.

However, Windows 11's new context menu _sucks_ with them.

**Comment by Somebodyisnobody** [comment]
I am aware that it depends on which arguments are delivered. But I see this as bad user experience when the user needs to open a command line to start his scenario. The concept `/network` etc. is in the responsibility of the engine. So I address this to the engine. It's a feature request to build in such a dialog:
![Dialog](https://user-images.githubusercontent.com/35230554/165276623-b091d7dd-e2a3-421c-a6c1-33a29e2525bb.png)



> However, Windows 11's new context menu _sucks_ with them.
👍

**Comment by maxmitti** [comment]
What if opening a scenario via the shell just launches it with `/lobby /nosignup`?
Technically its always a network game then, but it does not really make much of a difference anyway.
The benefit of the approach is that the user can then choose to turn on masterserver registration as usual from within the lobby, if desired, or just start the game without registering with the masterserver, while also being able to adjust which players should join locally.

**Comment by Somebodyisnobody** [comment]
If you look

**Comment by Somebodyisnobody** [comment]
And I would like to give a concrete use case: scenarios that are not in the clonk folder, but e.g. in the download directory, because you have to quickly host the hastily downloaded scenario from ccan.de, as the network code at the other host is once again causing problems.

(Problems from a friend 😉)

**Comment by maxmitti** [comment]
> I'm just talking about displaying the dialig if clonk.exe is called with the c4Scenario as only argument. That's the case when double clicking on it.

Yes, that’s basically what I am talking about. Although I thought that the file association can include arguments. And in that case I would specify the arguments there as `/nosignup /lobby`.

**Comment by Somebodyisnobody** [comment]
So you mean the file extention association in the windows registry for example? And instead a dialog there's always a lobby without masterserver registration opened?
Yeah that would be an option but nevertheless it's a network game. The game is still reachable from local network maybe with runtimejoin. In the most cases it will not be an issue but in my mind it's just a workaround. On the other hand a positive aspect is that the user can choose his players.

I would be okay with both options

**Comment by Somebodyisnobody** [comment]
@Fulgen301 is working on this at https://github.com/legacyclonk/LegacyClonk/tree/shell which would also solve the issue at it's best way.
